src/runtime/types.d.ts

```diff
@@ -26,15 +26,15 @@ type BeforeLanguageSwitchHandler = (
   newLocale: string,
   initialSetup: boolean,
   context: NuxtApp
-) => string | void
+) => Promise<unknown>
```

Comment:
```suggestion
) => Promise<string | void>
```

Reply:
I'm not sure if the string can be returned from the hook. There would be a minor workaround required from developers to assign the locale instead of returning it.